getAlby/lightning-browser-extension
# feat: add opt-out of new signmessage

#### Link this PR to an issue
Fixes #629

#### Type of change (Remove other not matching type)

- Breaking change (fix or feature that would cause existing functionality to not work as expected)
Users can opt-out of the new login method by checking a setting.

#### Describe the changes you have made in this PR -

- A new "private key" format for LNDhub connectors
- Hash the message that we need to sign in both LNBits and LNDhub connectors.
- Add a toggle in the Settings so people can sign in the old way and recover their old accounts.

#### Screenshots of the changes (If any) -
![Schermafbeelding 2022-03-30 om 10 33 41](https://user-images.githubusercontent.com/33457577/160788127-f559b851-68db-4a72-89b6-f52257ce1f2c.png)


#### How Has This Been Tested?

- Check out the master branch of the extension and go to https://lnurl.fiatjaf.com
- Make sure that you have a LNDhub wallet as your current wallet in Alby.
- Click on LNURL Auth QR code
- Make a note of the `key` value. This is your identity on this site.
- Now, check out this PR, reload the extension, refresh the page and do this again. Your `key` should be different.
- Toggle the setting to use the legacy LNURL auth. Your `key` should be the same again.

To test that it is compatible with Bluewallet, use Bluewallet connected to an Alby account to login to https://lnurl.fiatjaf.com. Make a not of the public key. This should be the same key as the one that you get by logging in with Alby, if you use the new login and are connected to the same account. 

Alternatively, go to a site where you have logged in with Alby like lnmarkets.com or stacker.news. If you log in with the new method, you should have a fresh, empty account. If you toggle the legacy LNURL auth setting, you should have your old account back.

TODO: new key format for LNBits? @bumi ?
